From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mender Community <community@mender.io>
Date: Mon, 22 Sep 2026 00:00:00 +0000
Subject: [PATCH] configs toradex board specific mender integration

Move the U-Boot environment into the eMMC user area and enable a
redundant environment so that Mender can update the bootloader
environment atomically. apalis-imx8 already enables CONFIG_BOOTCOUNT_ENV
and CONFIG_BOOTCOUNT_LIMIT in its defconfig, so no header change is
required.

Signed-off-by: Mender Community <community@mender.io>
---
 configs/apalis-imx8_defconfig | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/configs/apalis-imx8_defconfig b/configs/apalis-imx8_defconfig
--- a/configs/apalis-imx8_defconfig
+++ b/configs/apalis-imx8_defconfig
@@ -6,8 +6,10 @@ CONFIG_SYS_MALLOC_F_LEN=0x4000
 CONFIG_NR_DRAM_BANKS=3
 CONFIG_HAS_CUSTOM_SYS_INIT_SP_ADDR=y
 CONFIG_CUSTOM_SYS_INIT_SP_ADDR=0x80200000
-CONFIG_ENV_SIZE=0x2000
-CONFIG_ENV_OFFSET=0xFFFFDE00
+CONFIG_ENV_SIZE=0x4000
+CONFIG_ENV_OFFSET=0x800000
+CONFIG_ENV_OFFSET_REDUND=0x1000000
+CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
 CONFIG_DM_GPIO=y
 CONFIG_DEFAULT_DEVICE_TREE="fsl-imx8qm-apalis"
 CONFIG_BOOTAUX_RESERVED_MEM_BASE=0x88000000
@@ -70,7 +72,7 @@ CONFIG_CMD_EXT4_WRITE=y
 CONFIG_ENV_OVERWRITE=y
 CONFIG_ENV_IS_IN_MMC=y
 CONFIG_SYS_RELOC_GD_ENV_ADDR=y
-CONFIG_SYS_MMC_ENV_PART=1
+CONFIG_SYS_MMC_ENV_PART=0
 CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG=y
 CONFIG_VERSION_VARIABLE=y
 CONFIG_NET_RANDOM_ETHADDR=y
-- 
2.44.3

